src/pages/ForgotPassword/forgotpassword.spec.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import RequestPasswordResetPage, {
+  validateEmail
+} from '@pages/ForgotPassword/index';
+import { BrowserRouter, Switch } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import {
+  render,
+  fireEvent,
+  waitForDomChange,
+  cleanup,
+  wait
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+const mockStore = configureMockStore([thunk]);
+let store;
+const initialSate = {
+  isPending: false,
+  isSuccess: false,
+  error: null,
+  message: null
+};
+
+const props = {
+  isLoading: false,
+  showSuccess: false,
+  errorMessage: null,
+  successMessage: null,
+  handleSubmit: jest.fn()
+};
+
+const renderWithEnzymes = state => {
+  store = mockStore({
+    authReducer: {
+      error: null,
+      isAuthenticated: false,
+      user: {},
+      status: 'rest'
+    },
+    resetPassword: {
+      ...state
+    }
+  });
+  return mount(
+    <Provider store={store}>
+      <BrowserRouter>
+        <Switch>
+          <RequestPasswordResetPage {...props} />
+        </Switch>
+      </BrowserRouter>
+    </Provider>
+  );
+};
+
+const renderWithRTL = state => {
+  store = mockStore({
+    authReducer: {
+      error: null,
+      isAuthenticated: false,
+      user: {},
+      status: 'rest'
+    },
+    resetPassword: {
+      ...state
+    }
+  });
+  return {
+    ...render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Switch>
+            <RequestPasswordResetPage {...props} />
+          </Switch>
+        </BrowserRouter>
+      </Provider>
+    ),
+    store
+  };
+};
+
+describe('Test forgot password page', () => {
+  afterEach(cleanup);
+
+  it('should have a email address field', () => {
+    const wrapper = renderWithEnzymes();
+    const field = wrapper.find('input').first();
+    expect(field.exists()).toBe(true);
+    expect(field.props().name).toEqual('email');
+  });
+
+  it('should have a submit button', () => {
+    const wrapper = renderWithEnzymes();
+    const button = wrapper.find('button').first();
+    expect(button.exists()).toBe(true);
```

Comment:
can we assert for the text on the button i.e what text content is written on the button?   "ByText find by element text content" so as to be sure its the same button.